src/shared/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useState, useEffect, useContext } from "react";
+import { useNavigate } from "react-router-dom";
+import { styled } from "@mui/system";
+import { green, red, orange } from "@mui/material/colors";
+import { UserData } from "../../context/UserDataContext";
+import { IconButton, Box } from "@mui/material";
+import ArrowBackIosIcon from "@mui/icons-material/ArrowBackIos";
+import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";
+import dayjs, { Dayjs } from "dayjs";
+
+const Frame = styled("div")({
+  width: "900px",
+  border: "1px solid lightgrey",
+  boxShadow: "2px 2px 2px #eee",
+  margin: "0 auto",
+});
+
+const Header = styled("div")(({ theme }) => ({
+  fontSize: "17px",
+  fontWeight: "bold",
+  padding: "10px 10px 5px 10px",
+  display: "flex",
+  justifyContent: "space-around",
+  alignItems: "center",
+  backgroundColor: theme.palette.primary.main,
+  color: "white",
+}));
+
+const Body = styled("div")({
+  width: "100%",
+  padding: "20px",
+  display: "flex",
+  flexWrap: "wrap",
+});
+
+const WeekDays = styled("div")(({ theme }) => ({
+  width: "14%",
+  height: "40px",
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+  cursor: "pointer",
+  margin: "1px",
+  color: theme.palette.primary.main,
+}));
+const Day = styled("button")(({ theme }) => ({
+  width: "14%",
+  height: "40px",
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+  cursor: "pointer",
+  margin: "1px",
+  color: theme.palette.primary.main,
+}));
+
+const HeaderBox = styled("div")(({ theme }) => ({
+  width: "900px",
+  margin: "0 auto",
+  height: "40px",
+  display: "flex",
+  alignItems: "center",
+}));
+
+const ColorPoint = styled("div")(({ theme }) => ({
+  width: "12px",
+  height: "12px",
+  borderRadius: "50px",
+  margin: "0px 7px 0px 30px",
+}));
+
+const Calendar = () => {
+  const DAYS = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
```

Comment:
it's better to move those consts out of the Calendar component to avoid recreation of those static values on every render.